src/goto-instrument/synthesizer/expr_enumerator.cpp

```diff
@@ -0,0 +1,375 @@
+/*******************************************************************\
+Module: Enumerator Interface
+Author: Qinheping Hu
+\*******************************************************************/
+
+#include "expr_enumerator.h"
+
+#include <util/simplify_expr.h>
+
+#include <langapi/language_util.h>
```

Comment:
Is this perhaps a debugging leftover?

Reply:
No, `exprt`s are sorted by comparing their string representation produced by `from_expr`.

Reply:
Ah! I suggest you use `format` instead (from `util/format_expr.h`).